rmw_freertps_cpp/test/test_serialization.cpp

```diff
@@ -0,0 +1,615 @@
+    msg.uint64_value.push_back(i + 12);
+    msg.string_array.push_back(std::string("hello #") + std::to_string(i));
+  }
+  msg.string_value = "world";
+}
+
+void validate_factory_message(const rmw_freertps_cpp::msg::PrimitivesUnbounded & msg)
+{
+  for (uint32_t i = 0; i < 10; ++i) {
+    EXPECT_EQ(msg.bool_value[i], i % 2 == 0);
+    EXPECT_EQ(msg.byte_value[i], i + 1);
+    EXPECT_EQ(msg.char_value[i], i + 2);
+    EXPECT_FLOAT_EQ(msg.float32_value[i], i + 3 + 0.01);
+    EXPECT_DOUBLE_EQ(msg.float64_value[i], i + 4 + 0.001);
+    EXPECT_EQ(msg.int8_value[i], -i + 5);
+    EXPECT_EQ(msg.uint8_value[i], i + 6);
+    EXPECT_EQ(msg.int16_value[i], -i + 7);
+    EXPECT_EQ(msg.uint16_value[i], i + 8);
+    EXPECT_EQ(msg.int32_value[i], -i + 9);
+    EXPECT_EQ(msg.uint32_value[i], i + 10);
+    EXPECT_EQ(msg.int64_value[i], -i + 11);
+    EXPECT_EQ(msg.uint64_value[i], i + 12);
+    EXPECT_EQ(msg.string_array[i], std::string("hello #") + std::to_string(i));
+  }
+  EXPECT_EQ(msg.string_value, "world");
+}
+
+void message_factory(rmw_freertps_cpp::msg::PrimitiveStaticArrays & msg)
+{
+  for (uint32_t i = 0; i < 3; ++i) {
+    msg.bool_value[i] = i % 2 == 0;
+    msg.byte_value[i] = i + 1;
+    msg.char_value[i] = i + 2;
+    msg.float32_value[i] = i + 3 + 0.01;
+    msg.float64_value[i] = i + 4 + 0.001;
+    msg.int8_value[i] = -i + 5;
+    msg.uint8_value[i] = i + 6;
+    msg.int16_value[i] = -i + 7;
+    msg.uint16_value[i] = i + 8;
+    msg.int32_value[i] = -i + 9;
+    msg.uint32_value[i] = i + 10;
+    msg.int64_value[i] = -i + 11;
+    msg.uint64_value[i] = i + 12;
+  }
+}
+
+void validate_factory_message(const rmw_freertps_cpp::msg::PrimitiveStaticArrays & msg)
+{
+  for (uint32_t i = 0; i < 3; ++i) {
+    EXPECT_EQ(msg.bool_value[i], i % 2 == 0);
+    EXPECT_EQ(msg.byte_value[i], i + 1);
+    EXPECT_EQ(msg.char_value[i], i + 2);
+    EXPECT_FLOAT_EQ(msg.float32_value[i], i + 3 + 0.01);
+    EXPECT_DOUBLE_EQ(msg.float64_value[i], i + 4 + 0.001);
+    EXPECT_EQ(msg.int8_value[i], -i + 5);
+    EXPECT_EQ(msg.uint8_value[i], i + 6);
+    EXPECT_EQ(msg.int16_value[i], -i + 7);
+    EXPECT_EQ(msg.uint16_value[i], i + 8);
+    EXPECT_EQ(msg.int32_value[i], -i + 9);
+    EXPECT_EQ(msg.uint32_value[i], i + 10);
+    EXPECT_EQ(msg.int64_value[i], -i + 11);
+    EXPECT_EQ(msg.uint64_value[i], i + 12);
+  }
+}
+
+void message_factory(rmw_freertps_cpp::msg::LargeArray & msg)
+{
+  for (uint32_t i = 0; i < 256; ++i) {
+    msg.static_values[i] = i;
+    msg.unbounded_values.push_back(i);
+  }
+}
+
+void validate_factory_message(const rmw_freertps_cpp::msg::LargeArray & msg)
+{
+  for (uint32_t i = 0; i < 256; ++i) {
+    EXPECT_EQ(msg.static_values[i], i);
+    EXPECT_EQ(msg.unbounded_values[i], i);
+  }
+}
+
+TEST(test_serialization, serialize_empty) {
+  uint8_t * buf = nullptr;
+  {
+    rmw_freertps_cpp::msg::Empty msg;
+    uint32_t size = 0;
+    uint32_t result = rmw_freertps_cpp::msg::typesupport_freertps_cpp::serialize_ros_msg(&msg, buf,
```

Comment:
Wrap after the open parenthesis and then the next line is only indented one level.

Same below.